pkg/cloud/api/resource_test.go

```diff
@@ -862,6 +879,73 @@ func TestResourceImpliedVersion(t *testing.T) {
 	}
 }
 
+func TestImpliedVersionForPlaceHolderType(t *testing.T) {
+	PlaceholderTypeBeta := func() (meta.Version, error) {
```

Comment:
I find this func definition indirection and naming to be really hard to understand. because the names don't mean anything. 

It also establishes a pattern that will be hard to maintain, which is that each new test case will need to have a func defined here and then the func referenced below in the test case itself. This makes it so the person writing the test will need to think of a name for the func and then also makes sure the test case matches the set of funcs at the top.

It's better to inline the function -- the name isn't very meaningful and I am also looking around to see if it is used in other places.

Specifically:

```
type iv interface { ImpliedVersion() (meta.Version, error) }

tcs := []struct {
		desc    string
		makeResource func() iv
		wantErr bool
		wantVer meta.Version
	}{
		{
			desc:    "imply ga, PlaceholderType alpha",
			test:    func() iv { return newTestResource[..., ..., ...](nil) }
			wantVer: meta.VersionGA,
		},
```

Reply:
done